### PR TITLE
Fixed issues with writing nodegraphs to file.

### PR DESCRIPTION
**Pull request by niklasharrysson**

### CODE REVIEW

File: libraries/stdlib/stdlib_ng.mtlx [hunk 1]
```diff
@@ -43,7 +43,7 @@
       <parameter name="frameoffset" type="integer" interfacename="frameoffset"/>
       <parameter name="frameendaction" type="string" interfacename="frameendaction"/>
     </image>
-    <output name="N_out_float" type="float" nodename="N_img_float"/>
+    <output name="out" type="float" nodename="N_img_float"/>
```
Comment (niklasharrysson) [inline libraries/stdlib/stdlib_ng.mtlx]: The output names did not match the corresponding nodedef in stdlib_defs, where the outputs are declared explicitly now.

File: source/MaterialXRuntime/RtFileIo.cpp [hunk 2]
```diff
@@ -124,22 +165,77 @@ namespace
...
+            const PvtPortDef* nodedefPort = nodedef->findPort(RtToken(nodePort->getName()));
+            if (!nodedefPort || nodedefPort->getType().str() != nodePort->getType())
+            {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    PvtDataHandle resolveNodeDef(const NodePtr& node, PvtStage* stage)
```
Comment (niklasharrysson) [inline source/MaterialXRuntime/RtFileIo.cpp]: This new logic will extend the search for nodedefs to include the nodedefs already loaded in the runtime stage. This is a big improvement as it means all nodedefs no longer need to be included in the MaterialX document in order for a file to be loaded. Libraries with nodedefs can now be pre-loaded and referenced before loading MaterialX files with content data.

File: source/MaterialXTest/Runtime.cpp [hunk 3]
```diff
@@ -654,6 +653,65 @@ TEST_CASE("Runtime: FileIo", "[runtime]")
...
+
+    // Read the saved file to another stage.
+    mx::RtStage anotherStage = mx::RtStage::createNew("another");
+    anotherStage.addReference(libStage.getObject());
+    fileIo.setObject(anotherStage.getObject());
+    fileIo.read(filename, searchPath);
+
+    // Save the re-read stage to file for inspection.
+    const mx::FilePath filename2 = graph.getName().str() + "_another_export.mtlx";
+    fileIo.write(filename2, &options);
```
Comment (niklasharrysson) [inline source/MaterialXTest/Runtime.cpp]: write -> read -> write, again, to make sure both read and write of the data works correctly.

File: source/MaterialXRuntime/RtFileIo.cpp [hunk 4]
```diff
@@ -386,6 +447,44 @@ namespace
...
+                throw ExceptionRuntimeError("Node '" + name.str() + "' has not been loaded in the stage");
+            }
+            return nodeH->asA<PvtNode>();
+        };
+        auto findPort = [](const RtToken& name, PvtNode* node)
+        {
+            RtPort port = node->findPort(name);
+            if (!port)
+            {
+                throw ExceptionRuntimeError("Node '" + node->getName().str() + "' has not port named '" + name.str() + "'");
```
Comment (feldstj) [inline source/MaterialXRuntime/RtFileIo.cpp]: "not" should be "no"

File: source/MaterialXTest/Runtime.cpp (hunk 3 above)
Comment (feldstj) [reply source/MaterialXTest/Runtime.cpp]: Could be wrong, but shouldn't you be verifying that the files are the same?

Comment (niklasharrysson) [reply source/MaterialXTest/Runtime.cpp]: I could, in this case I was lazy and just used eye balling to compare the files :)

Comment (niklasharrysson) [reply source/MaterialXTest/Runtime.cpp]: I can add an automatic compare.

**Pull request merged**